core/server/data/xml/sitemap/page-generator.js

```diff
@@ -26,6 +26,7 @@ _.extend(PageMapGenerator.prototype, {
             context: {
                 internal: true
             },
+            filter: 'visibility:public',
```

Comment:
Ah whoops, that's a 🐛 . I thought it was added in the base generator and the base generator always returns true for validations. Will fix.